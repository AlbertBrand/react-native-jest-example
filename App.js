/* @flow */
import React from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>
        Welcome fellow traveler!
      </Text>
      <Icon name="thumbs-up" size={64} color="#9ad3f1" />
      <Icon.Button name="facebook" backgroundColor="#3b5998">
        Login with Facebook
      </Icon.Button>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
});
