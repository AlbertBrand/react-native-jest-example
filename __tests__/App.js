/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('react-native-vector-icons/FontAwesome', () => {
  const React = require('react');
  const Icon = props => React.createElement('Icon', props, props.children);
  Icon.Button = 'Icon.Button';
  return Icon;
});

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />
    );
    expect(tree).toMatchSnapshot()
  });
})
