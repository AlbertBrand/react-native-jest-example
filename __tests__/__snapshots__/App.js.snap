// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome fellow traveler!
  </Text>
  <Icon
    color="#9ad3f1"
    name="thumbs-up"
    size={64}
  />
  <Icon.Button
    backgroundColor="#3b5998"
    name="facebook"
  >
    Login with Facebook
  </Icon.Button>
</View>
`;
